pkg/api/v1/types.go

```diff
@@ -1960,14 +1960,30 @@ type PodLogOptions struct {
 
 	// The container for which to stream logs. Defaults to only container if there is one container in the pod.
 	Container string `json:"container,omitempty"`
-
-	// Follow the log stream of the pod.
-	// Defaults to false.
+	// Follow the log stream of the pod. Defaults to false.
 	Follow bool `json:"follow,omitempty"`
-
-	// Return previous terminated container logs.
-	// Defaults to false.
+	// Return previous terminated container logs. Defaults to false.
 	Previous bool `json:"previous,omitempty"`
+	// A relative time duration to the current time from which to show logs. If this value
+	// preceeds the time a pod was started, only logs since the pod start will be returned.
+	// If this value is in the future, no logs will be returned.
+	// Only one of sinceDuration or sinceTimestamp may be specified.
+	SinceDuration string `json:"sinceDuration,omitempty"`
```

Comment:
I really hate (as a client author) doing time manipulation.  In fact,
getting time manipulation wrong is the most common client authors have.  I
understand the argument to use seconds only, so I won't fight hard for
duration, but for a naive client a duration is a much more valuable option.

On Wed, Sep 16, 2015 at 1:15 AM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/v1/types.go
> <https://github.com/kubernetes/kubernetes/pull/13780#discussion_r39594482>
> :
>
> >  	Previous bool `json:"previous,omitempty"`
> > +	// A relative time duration to the current time from which to show logs. If this value
> > +	// preceeds the time a pod was started, only logs since the pod start will be returned.
> > +	// If this value is in the future, no logs will be returned.
> > +	// Only one of sinceDuration or sinceTimestamp may be specified.
> > +	SinceDuration string `json:"sinceDuration,omitempty"`
>
> Duration in what units? time.Duration? Everywhere else in the API we
> currently use seconds.
>
> https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#naming-conventions
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13780/files#r39594482>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift